Lib/concurrent/futures/process.py

```diff
@@ -658,7 +660,8 @@ def _start_executor_manager_thread(self):
             self._executor_manager_thread = _ExecutorManagerThread(self)
             self._executor_manager_thread.start()
             _threads_wakeups[self._executor_manager_thread] = \
-                self._executor_manager_thread_wakeup
+                (self._shutdown_lock,
+                 self._executor_manager_thread_wakeup)
```

Comment:
Oh, I never tried a weakref.WeakKeyDictionary() where the value a tuple of two objects.

Reply:
Do you think this could be a problem? I did not even thing about this..

Reply:
As long as they key isn't a temporary object this is fine.